File: HapticFeedbackApp/customized-munim-bluetooth/ios/MunimBluetoothEventEmitter.m
//
//  MunimBluetoothEventEmitter.m
//  munim-bluetooth
//
//  Objective-C bridge for event emitter
//

#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(MunimBluetoothEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: HapticFeedbackApp/customized-munim-bluetooth/ios/MunimBluetoothEventEmitter.swift
//
//  MunimBluetoothEventEmitter.swift
//  munim-bluetooth
//
//  Event emitter for Bluetooth events
//

import Foundation
import React

@objc(MunimBluetoothEventEmitter)
class MunimBluetoothEventEmitter: RCTEventEmitter {
    
    public static var shared: MunimBluetoothEventEmitter?
    
    override init() {
        super.init()
        MunimBluetoothEventEmitter.shared = self
    }
    
    override func supportedEvents() -> [String]! {
        return [
            "deviceFound",
            "onDeviceFound",
            "scanResult",
            "connectionStateChanged",
            "characteristicValueChanged"
        ]
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    func emitDeviceFound(_ deviceData: [String: Any]) {
        sendEvent(withName: "deviceFound", body: deviceData)
    }
}
